diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/WhiteSpaceRule.kt

```diff
@@ -188,22 +188,28 @@ class WhiteSpaceRule(private val configRules: List<RulesConfig>) : Rule("horizon
                             it[1].elementType == LAMBDA_EXPRESSION &&
                             it[2].elementType == VALUE_ARGUMENT &&
                             // lambda is not passed as a named argument
-                            !it[2].hasChildOfType(EQ) &&
-                            // lambda is the first argument in the list
-                            it[2].prevSibling { prevNode -> prevNode.elementType == COMMA } == null
+                            !it[2].hasChildOfType(EQ)
                 }
                 ?: false
+
         val prevNode = whitespaceOrPrevNode.let { if (it.elementType == WHITE_SPACE) it.treePrev else it }
         val numWhiteSpace = whitespaceOrPrevNode.numWhiteSpaces()
         // note: the conditions in the following `if`s cannot be collapsed into simple conjunctions
         if (isFromLambdaAsArgument) {
-            if (numWhiteSpace != 0) {
+            val isFirstArgument = node
+                    .parents()
```

Comment:
```suggestion
                    .parent({ it.elementType == VALUE_ARGUMENT })
```
no need to analyze all parents here